src/ML/src/MANNAutoregressiveInputBuilder.cpp

```diff
@@ -0,0 +1,269 @@
+            return false;
+        }
+        if (param <= 0)
+        {
+            log()->error("{} The parameter {} cannot be negative.", logPrefix, paramName);
+            return false;
+        }
+        return true;
+    };
+
+    // get all the required parameters
+    bool ok = getParameter("base_vel_norm", m_pimpl->baseVelocityNorm);
+    ok = ok && getParameter("ellipsoid_forward_axis", m_pimpl->ellipsoidForwardAxis);
+    ok = ok && getParameter("ellipsoid_side_axis", m_pimpl->ellipsoidSideAxis);
+    ok = ok && getParameter("ellipsoid_backward_axis", m_pimpl->ellipsoidBackwardAxis);
+    ok = ok && getParameter("ellipsoid_scaling_factor", m_pimpl->ellipsoidScalingFactor);
+    ok = ok
+         && getParameter("max_facing_direction_angle_forward",
+                         m_pimpl->maxFacingDirectionAngleForward);
+    ok = ok
+         && getParameter("max_facing_direction_angle_backward",
+                         m_pimpl->maxFacingDirectionAngleBackward);
+    ok = ok
+         && getParameter("max_facing_direction_angle_side_opposite_sign",
+                         m_pimpl->maxFacingDirectionAngleSideOppositeSign);
+    ok = ok
+         && getParameter("max_facing_direction_angle_side_same_sign",
+                         m_pimpl->maxFacingDirectionAngleSideSameSign);
+    ok = ok && getParameter("number_of_knots", m_pimpl->numberOfKnots);
+    ok = ok && m_pimpl->curve.curve.initialize(ptr);
+
+    // if everything went fine we can resize the matrices
+    if (ok)
+    {
+        m_pimpl->output.desiredFutureBaseTrajectory.resize(2, m_pimpl->numberOfKnots);
+        m_pimpl->output.desiredFutureBaseVelocities.resize(2, m_pimpl->numberOfKnots);
+        m_pimpl->output.desiredFutureFacingDirections.resize(2, m_pimpl->numberOfKnots);
+
+        m_pimpl->curve.initialPoint.setZero();
+        m_pimpl->frontalDirection << 1, 0;
+
+        m_pimpl->fsm = Impl::FSM::Initialized;
+    }
+
+    return ok;
+}
+
+bool MANNAutoregressiveInputBuilder::advance()
+{
+    constexpr auto logPrefix = "[MANNAutoregressiveInputBuilder::advance]";
+    if (m_pimpl->fsm == Impl::FSM::Idle)
+    {
+        log()->error("{} Please initialize the class before calling advance.", logPrefix);
+        return false;
+    }
+
+    m_pimpl->fsm = Impl::FSM::OutputInvalid;
+
+    // evaluate the
+
+    m_pimpl->curve.controlPoint[1] = 0;
+
+    if (m_pimpl->input.motionDirection[0] >= 0)
+    {
+        m_pimpl->curve.controlPoint[0]
+            = m_pimpl->ellipsoidScalingFactor
+              * std::min(m_pimpl->ellipsoidForwardAxis, m_pimpl->input.motionDirection[0]);
+        m_pimpl->curve.finalPoint[0] = m_pimpl->input.motionDirection[0] * m_pimpl->ellipsoidScalingFactor
+                        * m_pimpl->ellipsoidForwardAxis;
+        m_pimpl->curve.finalPoint[1] = m_pimpl->input.motionDirection[1] * m_pimpl->ellipsoidScalingFactor
+                        * m_pimpl->ellipsoidSideAxis;
```

Comment:
This part seems the same on both cases. You may want to move it outside the if for readability sake

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/commit/e335c02a26e55760544e2d69ea1b686e753d4c7d